src/main/java/org/embulk/output/mailchimp/MailChimpOutputPluginDelegate.java

```diff
@@ -0,0 +1,144 @@
+import org.embulk.spi.Exec;
+import org.embulk.spi.Schema;
+import org.slf4j.Logger;
+
+import java.util.List;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static org.embulk.output.mailchimp.validation.ColumnDataValidator.checkExistColumns;
+
+/**
+ * Created by thangnc on 4/14/17.
+ */
+public class MailChimpOutputPluginDelegate
+        implements RestClientOutputPluginDelegate<MailChimpOutputPluginDelegate.PluginTask>
+{
+    private static final Logger LOG = Exec.getLogger(MailChimpOutputPluginDelegate.class);
+
+    public MailChimpOutputPluginDelegate()
+    {
+    }
+
+    public interface PluginTask
+            extends RestClientOutputTaskBase
+    {
+        @Config("maximum_retries")
+        @ConfigDefault("6")
+        int getMaximumRetries();
+
+        @Config("initial_retry_interval_millis")
+        @ConfigDefault("1000")
+        int getInitialRetryIntervalMillis();
+
+        @Config("maximum_retry_interval_millis")
+        @ConfigDefault("32000")
+        int getMaximumRetryIntervalMillis();
+
+        @Config("timeout_millis")
+        @ConfigDefault("60000")
+        int getTimeoutMillis();
+
+        @Config("auth_method")
+        @ConfigDefault("access_token")
+        AuthMethod getAuthMethod();
+
+        @Config("apikey")
+        @ConfigDefault("null")
+        Optional<String> getApikey();
+
+        @Config("access_token")
+        @ConfigDefault("null")
+        Optional<String> getAccessToken();
+
+        @Config("list_id")
+        String getListId();
+
+        @Config("merge_fields")
+        @ConfigDefault("null")
+        Optional<List<String>> getMergeFields();
+
+        @Config("interest_categories")
+        @ConfigDefault("null")
+        Optional<List<String>> getInterestCategories();
+
+        @Config("update_existing")
+        @ConfigDefault("false")
+        boolean getUpdateExisting();
+    }
+
+    /**
+     * Override @{@link RestClientOutputPluginDelegate#validateOutputTask(RestClientOutputTaskBase, Schema, int)}
+     * This method not only validates required configurations but also validates required columns
+     *
+     * @param task
+     * @param schema
+     * @param taskCount
+     */
+    @Override
+    public void validateOutputTask(final PluginTask task, final Schema schema, final int taskCount)
+    {
+        if (task.getAuthMethod() == AuthMethod.OAUTH) {
+            if (!task.getAccessToken().isPresent() || isNullOrEmpty(task.getAccessToken().get())) {
+                throw new ConfigException("'access_token' is required when auth_method is 'oauth'");
+            }
+        }
+        else if (task.getAuthMethod() == AuthMethod.API_KEY) {
+            if (!task.getApikey().isPresent() || isNullOrEmpty(task.getApikey().get())) {
+                throw new ConfigException("'apikey' is required when auth_method is 'api_key'");
+            }
+        }
+
+        if (isNullOrEmpty(task.getListId())) {
```

Comment:
AFAIK, `list_id` can be empty or null. Plugin does not know it can receive correct param or not